confluent-schema-registry.spec.in

```diff
@@ -21,7 +21,41 @@ Schema Registry provides a serving layer for your metadata. It provides a RESTfu
 
 %pre
 
-exit 0
+_user=cp-schema-registry
+_group=confluent
+
+getent group $_group 2>&1 >/dev/null || groupadd -r $_group
+getent passwd $_user 2>&1 >/dev/null || \
+    useradd -r -g $_group --home-dir /tmp --no-create-home -s /sbin/nologin -c "Confluent Schema Registry" $_user
+
+_permwarn=
+for dir in /var/log/confluent ; do
```

Comment:
It looks like we have branches in this loop covering cases other than the one in this list. Is this supposed to cover things like `/etc/schema-registry` to make that at least readable by the right group? Should that be listed here? What about the `/usr/share/java/schema-registry` and `/usr/share/java/kafka-serde-tools` directories?

Reply:
This is a common piece I copied around for both debian and rpm builds for a bunch of different repos, thus the verboseness for a single directory here, where in other places it actually sets up a handful. This is for writable directories: log and data dirs.

The main reason to do this in postinst rather than a proper %files RPM spec target is to avoid breaking existing installations.
We try to set up the directories to our liking, but we will not modify existing directories - the user will be instructed to do so at her own discretion.

For read-only directories (/usr/share, et.al) the standard install method and permissions are used (root:root).

Reply:
re: boilerplate, makes sense, especially now that i've seen the prs for other repos too.

is root:root what we want now though? or is this also a compatibility thing? only reason it was root:root originally is b/c we didn't have any other user.

Reply:
root:root seems to be the default permissions for official debian and rpm packages, regardless of run-user, so we should probably stick to that.

Reply:
huh, learn something new every day. sorta surprising as i figured debian people wouldn't rely on root.